models/tasks.js

```diff
@@ -105,6 +105,32 @@ const fetchTask = async (taskId) => {
   }
 };
 
+/**
+ * Fetch a task against the IssueId
+ * @param issueId { number }: issueId which will be used to fetch the task
+ * @return {Promise<taskData|Object>}
+ */
+const fetchTaskByIssueId = async (issueId) => {
+  try {
+    const task = await tasksModel.where("github.issue.id", "==", issueId).get();
+    const [taskDoc] = task.docs;
+    let updatedTaskData;
+    if (taskDoc) {
+      updatedTaskData = { id: taskDoc.id, ...taskDoc.data() };
+    }
+    const taskData = await fromFirestoreData(updatedTaskData);
+
+    if (taskData?.status) {
+      taskData.status = TASK_STATUS[taskData.status.toUpperCase()] || task.status;
```

Comment:
I added this as a fallback if the key is not a property of TASK_STATUS. But I think it's not required since the task status is set from a predefined list of status.

Reply:
Done.